frontend/public/components/secrets/create-secret.tsx

```diff
@@ -0,0 +1,155 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import { k8sCreate, k8sUpdate, K8sResourceKind } from '../../module/k8s';
+import { ButtonBar, Firehose, history, kindObj, StatusBox } from '../utils';
+import { getActiveNamespace, formatNamespacedRouteForResource, UIActions } from '../../ui/ui-actions';
+import { SafetyFirst } from '../safety-first';
+import { WebHookSecretSubform } from './subforms';
+
+const determineSecretTypeAbstraction = (secret) => {
+  const { data } = secret;
+  if (_.has(data, 'WebHookSecretKey')) {
```

Comment:
Style nit, but I think you will see something like:

```JavaScript 
return _.has(data, 'WebHookSecretKey') 
  ? 'webhook'
  : 'generic';
```
Throughout the app.  If possible I think we should try to conform.